turbopack/crates/turbopack/src/lib.rs

```diff
@@ -702,22 +660,24 @@ impl AssetContext for ModuleAssetContext {
         self: Vc<Self>,
         result: Vc<ResolveResult>,
         reference_type: Value<ReferenceType>,
+        ignore_unknown: bool,
```

Comment:
I would remove that and instead use `ModuleResolveResultItem::Unknown`.

`ModuleResolveResult::primary_modules_iter`, `ModuleResolveResult::first_module`, `ReferencedAsset::from_resolve_result`, and `to_single_pattern_mapping` could handle that `ModuleResolveResultItem::Unknown` and it probably should also handle `ModuleResolveResultItem::Error` which we don't handle correctly currently.